### PR TITLE
Add negative native module tests

### PR DESCRIPTION
**Pull request by saxena-anurag**
## Description

This PR adds negative native module tests:
1. Negative tests for invalid IOCTLs from UM.
2. Negative tests for loading invalid programs.

Also fixes a bug caught by the new tests.

## Testing

Added new tests.

## Documentation

NA

### CODE REVIEW

File: tests/end_to_end/end_to_end.cpp [hunk 1]
```diff
@@ -1998,3 +2006,197 @@ _map_reuse_3_test(ebpf_execution_type_t execution_type)
...
+    _create_service_helper(L"fake_program.dll", NATIVE_DRIVER_SERVICE_NAME, &provider_module_id, &service_handle);
+
+    // Load native module. It should fail.
+    service_path = service_path + NATIVE_DRIVER_SERVICE_NAME;
+    REQUIRE(
+        test_ioctl_load_native_module(service_path, &provider_module_id, &count_of_maps, &count_of_programs) ==
+        ERROR_PATH_NOT_FOUND);
+}
+
+// Load native module by passing invalid service name to EC
```
Comment (dthaler) [inline tests/end_to_end/end_to_end.cpp:2054]:
```suggestion
// Load native module by passing invalid service name to EC.
```

File: tests/end_to_end/test_helper.cpp [hunk 2]
```diff
@@ -238,7 +240,11 @@ static void
 _preprocess_load_native_module(_Inout_ service_context_t* context)
 {
     context->dll = LoadLibraryW(context->file_path.c_str());
-    REQUIRE(context->dll != nullptr);
+    REQUIRE((context->dll != nullptr || _expect_native_module_load_failures));
```
Comment (dthaler) [inline tests/end_to_end/test_helper.cpp:243]:
```suggestion
    REQUIRE((context->dll != nullptr) || (_expect_native_module_load_failures));
```

File: tests/libs/util/ioctl_helper.cpp [hunk 3]
```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+// #include <chrono>
+// #include <filesystem>
```
Comment (dthaler) [inline tests/libs/util/ioctl_helper.cpp:4]:
```suggestion
```

remove commented out lines

File: tests/sample/unsafe/invalid_maps2.c [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+#define BPF_MAP_TYPE_INVALID 100
+
+typedef struct _process_entry
+{
+    uint32_t count;
+    uint8_t name[64];
+} process_entry_t;
+
+// This invalid program creates  map with invalid map type.
+
```
Comment (dthaler) [inline tests/sample/unsafe/invalid_maps2.c:36]: Duplicate of line 19
```suggestion
```

**Review by dthaler — APPROVED** [review]

File: tests/libs/util/ioctl_helper.cpp [hunk 5]
```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+#include <future>
+#include <map>
+using namespace std::chrono_literals;
+
+#include "api_common.hpp"
+#include "catch_wrapper.hpp"
+#include "ebpf_protocol.h"
+#include "device_helper.hpp"
```
Comment (dthaler) [inline tests/libs/util/ioctl_helper.cpp:10]:
```suggestion
#include "device_helper.hpp"
#include "ebpf_protocol.h"
```

**Review by dthaler — APPROVED** [review]

**Review by dthaler — DISMISSED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Pull request merged**